spec/handlers/newflow/student_signup_spec.rb

```diff
@@ -69,14 +69,76 @@ module Newflow
             hash_including({ email_address: an_instance_of(EmailAddress) })
           )
         )
-        result
+        subject
       end
 
       it 'stores selection in User whether to receive newsletter or not' do
         expect(User.new.receive_newsletter).to be_falsey
-        result
+        subject
         expect(User.last.receive_newsletter).to be(true)
       end
+
+      it 'outputs a user' do
+        expect(subject.outputs.user).to be_present
+      end
+    end
+
+    context 'success -- when passed-in a user_from_signed_params' do
+      # a user that already had an account, then tries to sign up using signed params (so via an LMS),
+      # should be successful.
+
+      before(:all) do
+        load('db/seeds.rb')
+      end
+
+      let(:email) do
+        Faker::Internet.free_email
+      end
+
+      let(:signed_params) do
+        {
+          uuid: Faker::Internet.uuid,
+          email: email,
+          school: 'some school',
+          role: 'student',
+          name: 'Bryan Dimas'
+          # signed_external_data: {  },
+        }.with_indifferent_access
+      end
+
+      let(:params) do
+        {
+          signup: {
+            first_name: 'Bryan',
+            last_name: 'Dimas',
+            email: email,
+            school: 'some school',
+            password: Faker::Internet.password(min_length: 8),
+            terms_accepted: true,
+            newsletter: true,
+            contract_1_id: FinePrint::Contract.first.id,
+            contract_2_id: FinePrint::Contract.second.id,
+            role: :student
+          }
+        }
+      end
+
+      let(:user_from_signed_params) do
+        Newflow::FindOrCreateUserFromSignedParams.call(signed_params).outputs.user
+      end
+
+      let(:subject) do
+        described_class.call(params: params, user_from_signed_params: user_from_signed_params)
+      end
+
+      it 'outputs a user' do
+        expect(subject.outputs.user).to be_present
+      end
+
+      it 'only creates one user' do
+        subject
```

Comment:
subject is the object under test, it should do something.  e.g.,  signed_in_user.call (where signed_in_user is your subject).

Reply:
It **does** something :)
```
        described_class.call(params: params, user_from_signed_params: user_from_signed_params)
```

Reply:
@BryanEliDimas sorry, that wasnt my point.  my point was that the actual subject is an object (not an object and a verb).  see "RSpec's subject is a special variable that refers to the object being tested"https://stackoverflow.com/questions/38437162/whats-the-difference-between-rspecs-subject-and-let-when-should-they-be-used

Reply:
Isn't the `subject` the `described_class`, then?

Reply:
@BryanEliDimas yeah, if it's that simple. sure.   the overall point is to use a named subject when it adds "documentation" and clarity to the test.  but of course, not to overengineer it, def didnt mean that :)